docs/persistence.rst

```diff
@@ -103,6 +103,148 @@ in any of the `range fields
     # empty range is unbounded
     Range().lower # None, False
 
+Python Type Hints
+~~~~~~~~~~~~~~~~~
+
+Document fields can be defined using standard Python type hints if desired.
+Here are some simple examples:
+
+.. code:: python
+
+    from typing import Optional
+
+    class Post(Document):
+        title: str                      # same as Text(required=True)
+        created_at: Optional[datetime]  # same as Date(required=False)
+        published: bool                 # same as Boolean(required=True)    
```

Comment:
Would it be helpful to call out that using type hints has different syntax than assignments somewhere?

Reply:
There is an error at runtime if you mistakenly use a DSL type as a type hint. If you do this:

```python
class MyDoc(Document):
    title: Text
```

You get this error:

```
TypeError: Cannot map type <class 'elasticsearch_dsl.field.Text'>
```

And if you do this:

```python
class MyDoc(Document):
    title: Text(required=True)
```

The error is:

```
TypeError: Cannot map type Text()
```